samples/boards/cc3220sf_launchxl/echo/src/simplelink_support.c

```diff
@@ -0,0 +1,535 @@
+			    netapp_event->Id);
+		break;
+	}
+}
+
+/**
+ *  @brief SimpleLinkGeneralEventHandler
+ *
+ *  This handler gets called whenever a general error is reported
+ *  by the NWP / Host driver. Since these errors are not fatal,
+ *  the application can handle them.
+ *
+ *  @note See the CC3120/CC3220 NWP programmer's guide (SWRU455)
+ *	  section 17.9.
+ */
+void SimpleLinkGeneralEventHandler(SlDeviceEvent_t *dev_event)
+{
+	if (!dev_event) {
+		return;
+	}
+
+	SYS_LOG_INF("\n[GENERAL EVENT] - ID=[%d] Sender=[%d]",
+		    dev_event->Data.Error.Code,
+		    dev_event->Data.Error.Source);
+}
+
+/**
+ *  @brief SimpleLinkFatalErrorEventHandler
+ *
+ *  This handler gets called whenever a driver error occurs requiring
+ *  restart of the device in order to recover.
+ */
+void SimpleLinkFatalErrorEventHandler(SlDeviceFatal_t *fatal_err_event)
+{
+
+	switch (fatal_err_event->Id) {
+	case SL_DEVICE_EVENT_FATAL_DEVICE_ABORT:
+		SYS_LOG_ERR("\n[ERROR] - FATAL ERROR: "
+			    "Abort NWP event detected: "
+			    "AbortType=%ld, AbortData=0x%lx",
+			    fatal_err_event->Data.DeviceAssert.Code,
+			    fatal_err_event->Data.DeviceAssert.Value);
+		break;
+
+	case SL_DEVICE_EVENT_FATAL_DRIVER_ABORT:
+		SYS_LOG_ERR("\n[ERROR] - FATAL ERROR: Driver Abort detected.");
+		break;
+
+	case SL_DEVICE_EVENT_FATAL_NO_CMD_ACK:
+		SYS_LOG_ERR("\n[ERROR] - FATAL ERROR: No Cmd Ack detected "
+			    "[cmd opcode = 0x%lx]",
+			    fatal_err_event->Data.NoCmdAck.Code);
+		break;
+
+	case SL_DEVICE_EVENT_FATAL_SYNC_LOSS:
+		SYS_LOG_ERR("\n[ERROR] - FATAL ERROR: Sync loss detected");
+		break;
+
+	case SL_DEVICE_EVENT_FATAL_CMD_TIMEOUT:
+		SYS_LOG_ERR("\n[ERROR] - FATAL ERROR: "
+			    "Async event timeout detected "
+			    "[event opcode =0x%lx]",
+			    fatal_err_event->Data.CmdTimeout.Code);
+		break;
+
+	default:
+		SYS_LOG_ERR("\n[ERROR] - FATAL ERROR: "
+			    "Unspecified error detected");
+		break;
+	}
+}
+
+/* Unused, but must be defined to link.	 */
+void SimpleLinkSockEventHandler(SlSockEvent_t *psock)
+{
+	ARG_UNUSED(psock);
+}
+
+/* Unused, but must be defined to link.	 */
+void SimpleLinkHttpServerEventHandler(SlNetAppHttpServerEvent_t *http_event,
+				      SlNetAppHttpServerResponse_t *http_resp)
+{
+	ARG_UNUSED(http_event);
+	ARG_UNUSED(http_resp);
+}
+
+
+/* Unused, but must be defined to link.	 */
+void SimpleLinkNetAppRequestEventHandler(SlNetAppRequest_t *netapp_request,
+					 SlNetAppResponse_t *netapp_response)
+{
+	ARG_UNUSED(netapp_request);
+	ARG_UNUSED(netapp_response);
+}
+
+/* Unused, but must be defined to link.	 */
+void SimpleLinkNetAppRequestMemFreeEventHandler(u8_t *buffer)
+{
+	ARG_UNUSED(buffer);
+}
+
+static long wlan_connect(void)
+{
+	SlWlanSecParams_t secParams = { 0 };
+	long lretval = 0;
+
+	secParams.Key = (signed char *)SECURITY_KEY;
+	secParams.KeyLen = strlen(SECURITY_KEY);
+	secParams.Type = SECURITY_TYPE;
+
+	lretval = sl_WlanConnect((signed char *)SSID_NAME,
+				 strlen(SSID_NAME), 0, &secParams, 0);
```

Comment:
This is bogus, SSID are not character arrays. You could have a '\0' in the middle and that would be just valid according to the specs.
SSID have to be treated as byte arrays (and thus '\0' cannot be used to know the end of the array).

Reply:
OK, thanks for pointing that out.   I changed to sizeof(SSID_NAME) - 1, since SSID_NAME is defined in an environment variable or preprocessor macro as a string, there is still the terminating null character to remove.    I also checked that sl_WlanConnect() is copying a byte array of the length argument passed in, and checking against the max length (32) there.